tests/unit/test_charm.py

```diff
@@ -2,8 +2,275 @@
+            shell=True,
+            stdout=-1,
+            encoding="utf-8",
+        )
+
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("os.mkdir")
+    @patch("shutil.copy")
+    @patch("shutil.rmtree")
+    @patch("subprocess.run")
+    def test_given_install_event_when_install_then_srs_directories_are_removed_and_recreated(
+        self, _, patch_rmtree, __, patch_mkdir, ___
+    ):
+        self.harness.charm.on.install.emit()
+
+        rmtree_calls = [
+            call("/srsLTE", ignore_errors=True),
+            call("/build", ignore_errors=True),
+            call("/config", ignore_errors=True),
+            call("/service", ignore_errors=True),
+        ]
+        mkdir_calls = [call("/srsLTE"), call("/build"), call("/config"), call("/service")]
+        patch_rmtree.assert_has_calls(calls=rmtree_calls)
+        patch_mkdir.assert_has_calls(calls=mkdir_calls)
+
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("os.mkdir")
+    @patch("shutil.copy")
+    @patch("shutil.rmtree")
+    @patch("subprocess.run")
+    def test_given_install_event_when_install_then_srsran_repo_is_cloned(
+        self, patch_subprocess_run, _, __, ___, ____
+    ):
+        self.harness.charm.on.install.emit()
+
+        patch_subprocess_run.assert_any_call(
+            "git clone --branch=release_20_10 --depth=1 https://github.com/srsLTE/srsLTE.git /srsLTE",  # noqa: E501
+            shell=True,
+            stdout=-1,
+            encoding="utf-8",
+        )
+
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("os.mkdir")
+    @patch("shutil.copy")
+    @patch("shutil.rmtree")
+    @patch("subprocess.run")
+    def test_given_install_event_when_install_then_srsran_is_built(
+        self, patch_subprocess_run, _, __, ___, ____
+    ):
+        self.harness.charm.on.install.emit()
+
+        patch_subprocess_run.assert_any_call(
+            "cd /build && cmake /srsLTE && make -j `nproc` srsenb srsue",
+            shell=True,
+            stdout=-1,
+            encoding="utf-8",
+        )
+
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("os.mkdir")
+    @patch("shutil.copy")
+    @patch("shutil.rmtree")
+    @patch("subprocess.run")
+    def test_given_install_event_when_install_then_files_are_copied(
+        self, _, __, patch_copy, ___, ____
+    ):
+        self.harness.charm.on.install.emit()
+
+        calls = [
+            call("/srsLTE/srsenb/enb.conf.example", "/config/enb.conf"),
+            call("/srsLTE/srsenb/drb.conf.example", "/config/drb.conf"),
+            call("/srsLTE/srsenb/rr.conf.example", "/config/rr.conf"),
+            call("/srsLTE/srsenb/sib.conf.example", "/config/sib.conf"),
+            call("/srsLTE/srsenb/sib.conf.mbsfn.example", "/config/sib.mbsfn.conf"),
+            call("/srsLTE/srsue/ue.conf.example", "/config/ue.conf"),
```

Comment:
We could generate the list of expected calls leveraging the constants from the charm. That way we'd stay in sync with the charm even if it changes.
WDYT?

Reply:
Again, I disagree